fe/fe-core/src/main/java/com/starrocks/lake/snapshot/ClusterSnapshotCheckpointScheduler.java

```diff
@@ -0,0 +1,166 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.lake.snapshot;
+
+import com.starrocks.common.Config;
+import com.starrocks.common.Pair;
+import com.starrocks.common.StarRocksException;
+import com.starrocks.common.util.FrontendDaemon;
+import com.starrocks.lake.snapshot.ClusterSnapshotJob.ClusterSnapshotJobState;
+import com.starrocks.leader.CheckpointController;
+import com.starrocks.server.GlobalStateMgr;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+// ClusterSnapshotCheckpointScheduler daemon is running on master node. Coordinate two checkpoint controller
+// together to finish image checkpoint one by one and upload image for backup
+public class ClusterSnapshotCheckpointScheduler extends FrontendDaemon {
+    public static final Logger LOG = LogManager.getLogger(ClusterSnapshotCheckpointScheduler.class);
+    private static int CAPTURE_ID_RETRY_TIME = 10;
+
+    private final CheckpointController feController;
+    private final CheckpointController starMgrController;
+
+    public ClusterSnapshotCheckpointScheduler(CheckpointController feController, CheckpointController starMgrController) {
+        super("cluster_snapshot_checkpoint_scheduler", Config.automated_cluster_snapshot_interval_seconds * 1000L);
+        this.feController = feController;
+        this.starMgrController = starMgrController;
+    }
+
+    @Override
+    protected void runAfterCatalogReady() {
+        if (!GlobalStateMgr.getCurrentState().getClusterSnapshotMgr().isAutomatedSnapshotOn()) {
+            return;
+        }
+
+        CheckpointController.exclusiveLock();
+        try {
+            runCheckpointScheduler();
+        } finally {
+            CheckpointController.exclusiveUnlock();
+        }
+    }
+
+    protected void runCheckpointScheduler() {
+        String errMsg = "";
+        ClusterSnapshotJob job = GlobalStateMgr.getCurrentState().getClusterSnapshotMgr()
+                                                                 .createAutomatedSnapshotJob(); /* INITIALIZING state */
+
+        do {
+            // step 1: capture consistent journal id for checkpoint
+            Pair<Long, Long> consistentIds = captureConsistentCheckpointIdBetweenFEAndStarMgr();
+            if (consistentIds == null) {
+                errMsg = "failed to capture consistent journal id for checkpoint";
+                break;
+            }
+            job.setJournalIds(consistentIds.first, consistentIds.second);
+            LOG.info("Successful capture consistent journal id, FE checkpoint journal Id: {}, StarMgr checkpoint journal Id: {}",
+                     consistentIds.first, consistentIds.second);
```

Comment:
Roll a new journal db after this step?

Reply:
It seems that it is hard to Roll a new journal in a sync way. Especially if no new editlog is comming

Reply:
OK